### PR TITLE
fix: live preview: added variant support in data cslp tag function

### PR DESCRIPTION
**Pull request by contentstackMridul**
fix: live preview: added variant support in data cslp tag function

### CODE REVIEW

File: src/live-preview.ts [hunk 1]
```diff
@@ -299,19 +308,25 @@ export default class LivePreview {
...
-                cslpTag.split(".");
-
+            const {
+                content_type_uid,
+                entry_uid,
+                locale,
+                variant,
+                fieldPath: field,
+            } = extractDetailsFromCslp(cslpTag);
+            console.log("🚀 ~ LivePreview ~ scrollHandler ~ variant:", variant);
```
Comment (KANE-99) [inline src/live-preview.ts:318]: Remove this console log @contentstackMridul

File: src/live-preview.ts [hunk 2]
```diff
@@ -323,7 +338,13 @@ export default class LivePreview {
                         content_type_uid,
                         locale,
                         entry_uid,
-                        field.join(".")
+                        variant,
+                        field
+                    );
+                    console.log(
```
Comment (KANE-99) [inline src/live-preview.ts:344]: remove this console @contentstackMridul

File: src/live-preview.ts (hunk 1 above)
Comment (contentstackMridul) [reply src/live-preview.ts]: done

File: src/live-preview.ts (hunk 2 above)
Comment (contentstackMridul) [reply src/live-preview.ts]: done

File: src/__test__/live-preview.test.ts [hunk 3]
```diff
@@ -222,6 +229,47 @@ describe("cslp tooltip", () => {
...
+        const hoverEvent = new CustomEvent("mouseover", {
+            bubbles: true,
+        });
+
+        variantTitlePara?.dispatchEvent(hoverEvent);
+
+        singularEditButton?.click();
+
+        const expectedRedirectUrl =
+            "https://app.contentstack.com/#!/stack/sample-api-key/content-type/content-type-1/en-us/entry/entry-uid-1/variant/variant-uid-1/edit?branch=dev&preview-field=field-title&preview-locale=en-us&preview-environment=sample-environment";
```
Comment (RavenColEvol) [inline src/__test__/live-preview.test.ts:266]: yeah ek config variable say hi derive karta tho better hota

**Review by RavenColEvol — APPROVED** [review]
LGTM

File: src/__test__/live-preview.test.ts (hunk 3 above)
Comment (contentstackMridul) [reply src/__test__/live-preview.test.ts:266]: I'll add it later

**Pull request merged**